internal/service/http/http.go

```diff
@@ -178,8 +178,9 @@ func (s *Service) Run(ctx context.Context, host service.Host) error {
 	}
 
 	// NOTE(@tpaschalis) These need to be kept in order for the longer
-	// remotecfg prefix to be invoked correctly.
-	r.PathPrefix(s.componentHttpPathPrefixRemotecfg).Handler(s.componentHandler(remoteCfgHostProvider(host), s.componentHttpPathPrefixRemotecfg))
+	// remotecfg prefix to be invoked correctly. The pathPrefix is still the
+	// same so that `remotecfg/` is not stripped from component lookups.
+	r.PathPrefix(s.componentHttpPathPrefixRemotecfg).Handler(s.componentHandler(remoteCfgHostProvider(host), s.componentHttpPathPrefix))
```

Comment:
if both calls to `s.componentHandler` set pathPrefix to the same value, we should probably not have it as a method parameter

Reply:
I think you're right. Do you mind if I follow up on this in a different PR? The parameter was already there so I want to see if there are any other potential uses for it.

Reply:
I don't really mind, although it should be quite easy to check cause it'd give you compilation error.